stac_fastapi/pgstac/tests/api/test_api.py

```diff
@@ -302,3 +316,62 @@ async def test_search_line_string_intersects(
     assert resp.status_code == 200
     resp_json = resp.json()
     assert len(resp_json["features"]) == 1
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("app_client", [{"setup_func": cors_disable}], indirect=True)
+async def test_without_cors(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_permit_origin})
+    assert resp.status_code == HTTPStatus.OK
+    assert (
+        len(
+            [
+                header
+                for header in resp.headers
+                if header.startswith("access-control-allow-")
+            ]
+        )
+        == 0
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("app_client", [{"setup_func": cors_enable}], indirect=True)
+async def test_with_match_cors(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_permit_origin})
+    assert resp.status_code == HTTPStatus.OK
+    assert resp.headers["access-control-allow-origin"] == cors_permit_origin
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("app_client", [{"setup_func": cors_enable}], indirect=True)
+async def test_with_mismatch_cors(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_deny_origin})
+    assert resp.status_code == HTTPStatus.OK
+    assert (
+        len(
+            [
+                header
+                for header in resp.headers
+                if header.startswith("access-control-allow-")
+            ]
+        )
+        == 0
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("app_client", [{"setup_func": cors_missing}], indirect=True)
+async def test_with_missing_config(app_client):
+    resp = await app_client.get("/", headers={"Origin": cors_permit_origin})
+    assert resp.status_code == HTTPStatus.OK
```

Comment:
If you're missing the config, you want this to pass with http 200?

Reply:
That's how CORS works - if a browser submits an HTTP request from a domain not listed in CORS the request is successful but does not include the right headers. The browser then prevents JavaScript from accessing data from the response. It's actually infuriating if you're a web developer as you can see the request succeed in your network traffic but then you don't get any data 🤷 . CORS is all about bypassing the browser's same-origin policy; it's not about server security or authorisation: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS